pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/impl/inv/OnHeapBitmapInvertedIndexCreator.java

```diff
@@ -45,35 +44,32 @@ public OnHeapBitmapInvertedIndexCreator(File indexDir, String columnName, int ca
   }
 
   @Override
-  public void addSV(int docId, int dictId) {
-    _bitmaps[dictId].add(docId);
+  public void add(int dictId) {
+    _bitmaps[dictId].add(_nextDocId++);
   }
 
   @Override
-  public void addMV(int docId, int[] dictIds) {
-    addMV(docId, dictIds, dictIds.length);
-  }
-
-  @Override
-  public void addMV(int docId, int[] dictIds, int numDictIds) {
-    for (int i = 0; i < numDictIds; i++) {
-      _bitmaps[dictIds[i]].add(docId);
+  public void add(int[] dictIds, int length) {
+    for (int i = 0; i < length; i++) {
+      _bitmaps[dictIds[i]].add(_nextDocId);
     }
+    _nextDocId++;
   }
 
   @Override
   public void seal() throws IOException {
     try (DataOutputStream out = new DataOutputStream(
         new BufferedOutputStream(new FileOutputStream(_invertedIndexFile)))) {
-      // Write all offsets
-      int offset = (_bitmaps.length + 1) * V1Constants.Numbers.INTEGER_SIZE;
-      out.writeInt(offset);
+      // Write bitmap offsets
+      int bitmapOffset = (_bitmaps.length + 1) * Integer.BYTES;
+      out.writeInt(bitmapOffset);
       for (MutableRoaringBitmap bitmap : _bitmaps) {
-        offset += bitmap.serializedSizeInBytes();
+        bitmapOffset += bitmap.serializedSizeInBytes();
         // Check for int overflow
-        Preconditions.checkState(offset > 0, "Inverted index file exceeds 2GB limit for column: %s", _columnName);
-        out.writeInt(offset);
+        Preconditions.checkState(bitmapOffset > 0, "Inverted index file: {} exceeds 2GB limit", _invertedIndexFile);
```

Comment:
This check is a bit confusing, esp given that its missing in the off-heap version. It appears that either its applicable in both places or neither. Let me know if I am missing something.

Reply:
Seems I accidentally removed it in the offHeap one. Thanks for catching this

Reply:
We should consider moving the part where we write the indices to file (and other common parts) into an abstract class - that way we can have common enforcement for such limits.

Reply:
They are written in different ways. For the on-heap one, we already get all bitmaps, but for the off-heap one, we build bitmaps on the fly, so we keep two file descriptors at the same time.It's hard to make them a common method.